R/bc_dunif.cpp

```diff
@@ -0,0 +1,81 @@
+#include <Rcpp.h>
+#include <RcppEigen.h>
+#include <cmath>
+
+// [[Rcpp::depends(RcppEigen)]]
+
+using Eigen::Map;                       // 'maps' rather than copies
+using Eigen::VectorXd;                  // variable size vector, double precision
+using Eigen::RowVectorXd;               // double precision row vector
+using Rcpp::NumericVector;
+
+
+// Bias correction function (Chen 1999)
+double rho(double x,  double b) {
+  double c1 = 2.0*pow(b,2.0) + 2.5;
+  double c2 = sqrt(4.0*pow(b, 4.0) + 6.0*pow(b, 2.0) + 2.25 - pow(x,2.0) - x/b);
+  return c1 + c2;
+}
+
+// Helper function to generate dbeta value vector 
+VectorXd vec_dbeta(Map<VectorXd> x, double a, double b){
```

Comment:
Does this have to be passed in by value? I am also not sure on the reasoning behind using a map. `VectorXd& x` would pass by reference.

Reply:
@rok-cesnovar Thanks for going over the code! I think I forgot to remove this function, this isn't currently used in the code. I hadn't been able to get `Rcpp::dbeta` (inherently works with vecs) to work so was using `R::dbeta` with this as a helper function.